fem/bilininteg_convection_pa.cpp

```diff
@@ -778,16 +893,83 @@ void ConvectionIntegrator::AssemblePA(const FiniteElementSpace &fes)
    const int nq = ir->GetNPoints();
    dim = mesh->Dimension();
    ne = fes.GetNE();
+   const DofToQuad::Mode mode = DofToQuad::TENSOR;
    geom = mesh->GetGeometricFactors(*ir, GeometricFactors::JACOBIANS);
-   maps = &el.GetDofToQuad(*ir, DofToQuad::TENSOR);
+   maps = &el.GetDofToQuad(*ir, mode);
    dofs1D = maps->ndof;
    quad1D = maps->nqpt;
    pa_data.SetSize(symmDims * nq * ne, Device::GetMemoryType());
    Vector vel;
-   if (VectorConstantCoefficient *cQ = dynamic_cast<VectorConstantCoefficient*>(Q))
+   if (VectorConstantCoefficient *cQ =
+          dynamic_cast<VectorConstantCoefficient*>(Q))
    {
       vel = cQ->GetVec();
    }
+   else if (VectorGridFunctionCoefficient *vgfQ =
+               dynamic_cast<VectorGridFunctionCoefficient*>(Q))
+   {
```

Comment:
Could the routine on lines ``908-972`` be done using or in ``QuadratureInterpolator``?  
It looks like you have a grid function and are sampling it at the quadrature points, is that right?  If so, I feel that routine could be useful in different parts of the code. The output could perhaps be wrapped in ``VectorQuadratureFunctionCoefficient``?  Mostly thinking of ways to make the code a bit more modular.

Reply:
It could be, sure. But we are splitting the PR, so we are going to create a dependency, but that's fine I guess.

Reply:
Waiting for #1737, then I'll push the code over there.

Reply:
`done`